src/solutions/solution_interface.jl

```diff
@@ -49,11 +49,12 @@ Base.@propagate_inbounds function Base.getindex(A::AbstractTimeseriesSolution,sy
       i = sym
   end
 
-  if i == nothing
-      if issymbollike(i) && Symbol(i) == getindepsym(A)
-        A.t
+    indepsym = getindepsym(A)
+    if i === nothing     
```

Comment:
why `===`? I don't think this could be inferred.

Reply:
I have used `===` whenever I compare against `nothing`. My understanding is that it does a type comparison in case of `nothing` -- am I correct to assume what you say implies that this only helps when the type of `i` is inferrable leads to the if the statement being compiled away? Does it cause any overhead if the type of `i` is not inferrable?

Reply:
It does a type-comparison, yes, which can be helpful for branch elimination. But `i` here is not known to be `nothing`, so we don't want to compare types I would think.

Reply:
https://docs.julialang.org/en/v1/manual/performance-tips/#Checking-for-equality-with-a-singleton This what I was intending to say.

Reply:
![image](https://user-images.githubusercontent.com/19674071/125312942-61a5a080-e352-11eb-9c76-aca10ff6a902.png)
I think the julia extension has also led to me subconsciously  always using `===` with `nothing`. 😅

Reply:
I have replace is it with normal euality.

Reply:
It's always recommended to use `===` to compare with nothing.